packages/styles/typography/_t.headings.scss

```diff
@@ -1,67 +1,82 @@
 /* mqp:start */
 .mt-heading {
-  &-l {
-    @include set-font-face('regular');
-    @include set-font-scale('09', 'm');
+  $parent: #{&};
 
-    @include set-from-screen('m') {
-      @include set-font-scale('10', 'm');
-    }
+  display: block;
 
-    &--semi-bold {
-      @include set-font-face('semi-bold');
-    }
+  // add underline style
+
+  &--underline::after {
+    content: '';
+    display: block;
+    height: $mu025;
+    background-color: $color-primary-01-500;
+  }
 
-    &--line-height-s {
-      @include set-line-height('09', 's');
+  // underline color variations
 
-      @include set-from-screen('m') {
-        @include set-line-height('10', 's');
-      }
-    }
+  &--line-primary-02::after {
+    background-color: $color-primary-02-600;
   }
 
-  &-m {
-    @include set-font-face('regular');
-    @include set-font-scale('08', 'm');
+  &--line-reverted::after {
+    background-color: $color-primary-01-200;
+  }
 
-    @include set-from-screen('m') {
-      @include set-font-scale('09', 'm');
-    }
+  &--line-primary-02-reverted::after {
+    background-color: $color-primary-02-200;
+  }
 
-    &--semi-bold {
-      @include set-font-face('semi-bold');
-    }
+  // text alignment
 
-    &--line-height-s {
-      @include set-line-height('08', 's');
+  &--center {
+    @include modify-from-screens($major-screens) {
+      text-align: center;
 
-      @include set-from-screen('m') {
-        @include set-line-height('09', 's');
+      &#{$parent}--underline::after {
+        margin-left: auto;
+        margin-right: auto;
       }
     }
   }
 
-  &-s {
-    @include set-font-face('regular');
-    @include set-font-scale('07', 'm');
-
-    @include set-from-screen('m') {
-      @include set-font-scale('08', 'm');
-    }
+  &--right {
+    text-align: right;
```

Comment:
I think you should include the `text-align: right` property, in the `modify-from-screens($major-screens)` mixin.